platform/android/java/build.gradle

```diff
@@ -163,6 +172,56 @@ def isAndroidStudio() {
     return sysProps != null && sysProps['idea.platform.prefix'] != null
 }
 
+task copyEditorDebugBinaryToBin(type: Copy) {
+    dependsOn ':editor:assembleDebug'
+    from('editor/build/outputs/apk/debug')
+    into(binDir)
+    include('android_editor_debug.apk')
+}
+
+task copyEditorRelease_debugBinaryToBin(type: Copy) {
+    dependsOn ':editor:assembleRelease_debug'
+    from('editor/build/outputs/apk/release_debug')
+    into(binDir)
+    include('android_editor_release_debug.apk')
+}
+
+task copyEditorReleaseBinaryToBin(type: Copy) {
+    dependsOn ':editor:assembleRelease'
+    from('editor/build/outputs/apk/release')
+    into(binDir)
+    include('android_editor_release.apk')
+}
```

Comment:
That doesn't match the build matrix for other platforms.

The official released (production) editor binaries use `target=release_debug`. The editor can't be used with `target=release` only, as debugging tools are then not included, and it would crash on errors instead of reporting them.

So the only two editor targets you should support are:
`tools=yes target=release_debug` -> production build, official releases
`tools=yes target=debug` -> development build, used by engine developers (slow and not meant for use in production)

Reply:
I have updated the logic to exclude the `release` target for editor builds and added a comment to explain the exclusion.

Reply:
We need to update the [documentation](https://docs.godotengine.org/en/stable/development/compiling/compiling_for_android.html#building-the-export-templates) for building export templates, because the default is now to build the editor version not the templates.

Reply:
Ah that default should be fixed IMO, I would expect it to still default to builtin templates.

I think it would still be the default for `scons` since there is:
```
def get_flags():
    return [
        ("tools", False),
    ]
```
which changes the default value for `tools` (but it can still be overridden with `tools=yes`).

So I guess the problem happens at the gradle step?

Reply:
Okay. So what's changed then is that the default target "debug" now builds a `dev.aar` instead of a `debug.aar`.

FYI: Previously I was using the following to build a quick test debug export template for android x86 emulators:
```
scons -j8 platform=android android_arch=x86
```
And then I run `./gradlew generateDevTemplate`
I now need to use:
```
scons -j8 target=release_debug platform=android android_arch=x86
```

Which matches the documentation, but seems less intuitive. I would expect `release_debug` to be for the editor builds  i.e. with `tools=yes`. If we're changing behaviour, it may be a good opportunity to change the names to match outputs and expectations.

Reply:
For `generateDevTemplate` is should be using `target=debug`, this might be a bug. `target=debug` is only used for development purposes and doesn't match the "Debug Template" that we build with `target=release_debug` (and which is the one `generateGodotTemplates` should use).

So it might have been an oversight among the other buildsystem changes.

> If we're changing behaviour, it may be a good opportunity to change the names to match outputs and expectations.

For `3.x` I think we should aim not to change behavior. For `master` I plan to implement https://github.com/godotengine/godot-proposals/issues/3371 which should help clean this up for Android too.